crates/bevy_transform/src/components/transform.rs

```diff
@@ -49,12 +56,9 @@ impl Transform {
     /// Creates a new identity [`Transform`], with no translation, rotation, and a scale of 1 on
     /// all axes.
     #[inline]
+    #[deprecated = "Use `Transform::IDENTITY` instead."]
     pub const fn identity() -> Self {
-        Transform {
-            translation: Vec3::ZERO,
-            rotation: Quat::IDENTITY,
-            scale: Vec3::ONE,
-        }
+        Transform::IDENTITY
```

Comment:
This can just be removed. Let's break things and move fast, and have less stuff to maintain.